data/transactions/verify/txn.go

```diff
@@ -418,6 +418,20 @@ func logicSigSanityCheckBatchPrep(txn *transactions.SignedTxn, groupIndex int, g
 	return nil
 }
 
+// LogicSigError represents a LogicSig evaluation which rejected or errored
+type LogicSigError struct {
```

Comment:
This new error type is introduced for two reasons:
* So the simulator can determine if the error is because of a rejecting/failing LogicSig eval. In that case, the simulation is still a "success" and no error is returned.
* This tells the simulator the group index of the txn with the failing LogicSig, so it can be reported properly in the results object.